RNTester/js/PanResponderExample.js

```diff
@@ -10,29 +10,27 @@
 
 'use strict';
 
-var React = require('react');
-var createReactClass = require('create-react-class');
-var ReactNative = require('react-native');
-var {PanResponder, StyleSheet, View} = ReactNative;
+const React = require('react');
+const ReactNative = require('react-native');
+const {PanResponder, StyleSheet, View} = ReactNative;
 
-var CIRCLE_SIZE = 80;
+const CIRCLE_SIZE = 80;
 
-var PanResponderExample = createReactClass({
-  displayName: 'PanResponderExample',
+class PanResponderExample extends React.Component<{}> {
+  static displayName: ?string = 'PanResponderExample';
 
   statics: {
     title: 'PanResponder Sample',
-    description:
-      'Shows the use of PanResponder to provide basic gesture handling.',
-  },
+    description: 'Shows the use of PanResponder to provide basic gesture handling.',
+  };
 
-  _panResponder: {},
-  _previousLeft: 0,
-  _previousTop: 0,
-  _circleStyles: {},
-  circle: (null: ?{setNativeProps(props: Object): void}),
+  _panResponder: Object = {};
+  _previousLeft: number = 0;
+  _previousTop: number = 0;
+  _circleStyles: Object = {};
```

Comment:
I think we should make this typing more specific. Something like: 

```typescript
import type {ViewStyleProp} from 'StyleSheet';

//...
  _circleStyles: {| style: ViewStyleProp |} = { style: {} } 
//...

Reply:
ViewStyleProp sis readonly type, so I think it is not correct....
In fact, I get many error...

```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ RNTester/js/PanResponderExample.js:58:30

Cannot assign 'blue' to this._circleStyles.style.backgroundColor because:
 • property backgroundColor is missing in $ReadOnlyArray [1].
 • property backgroundColor is missing in boolean literal false [1].
 • property backgroundColor is missing in empty string [1].
 • property backgroundColor is missing in null [1].
 • property backgroundColor is missing in undefined [1].
 • property backgroundColor is not writable.

 [1] 30│   _circleStyles: {|style: ViewStyleProp|} = {style: {}};
       :
     55│   }
     56│
     57│   _highlight = () => {
     58│     this._circleStyles.style.backgroundColor = 'blue';
     59│     this._updateNativeStyles();
     60│   };
     61│


Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ RNTester/js/PanResponderExample.js:63:30

Cannot assign 'green' to this._circleStyles.style.backgroundColor because:
 • property backgroundColor is missing in $ReadOnlyArray [1].
 • property backgroundColor is missing in boolean literal false [1].
 • property backgroundColor is missing in empty string [1].
 • property backgroundColor is missing in null [1].
 • property backgroundColor is missing in undefined [1].
 • property backgroundColor is not writable.

 [1] 30│   _circleStyles: {|style: ViewStyleProp|} = {style: {}};
       :
     60│   };
     61│
     62│   _unHighlight = () => {
     63│     this._circleStyles.style.backgroundColor = 'green';
     64│     this._updateNativeStyles();
     65│   };
     66│


Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ RNTester/js/PanResponderExample.js:92:30

Cannot assign this._previousLeft + gestureState.dx to this._circleStyles.style.left
because:
 • property left is missing in $ReadOnlyArray [1].
 • property left is missing in boolean literal false [1].
 • property left is missing in empty string [1].
 • property left is missing in null [1].
 • property left is missing in undefined [1].
 • property left is not writable.

 [1] 30│   _circleStyles: {|style: ViewStyleProp|} = {style: {}};
       :
     89│   };
     90│
     91│   _handlePanResponderMove = (e: Object, gestureState: Object) => {
     92│     this._circleStyles.style.left = this._previousLeft + gestureState.dx;
     93│     this._circleStyles.style.top = this._previousTop + gestureState.dy;
     94│     this._updateNativeStyles();
     95│   };


Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ RNTester/js/PanResponderExample.js:93:30

Cannot assign this._previousTop + gestureState.dy to this._circleStyles.style.top
because:
 • property top is missing in $ReadOnlyArray [1].
 • property top is missing in boolean literal false [1].
 • property top is missing in empty string [1].
 • property top is missing in null [1].
 • property top is missing in undefined [1].
 • property top is not writable.

 [1] 30│   _circleStyles: {|style: ViewStyleProp|} = {style: {}};
       :
     90│
     91│   _handlePanResponderMove = (e: Object, gestureState: Object) => {
     92│     this._circleStyles.style.left = this._previousLeft + gestureState.dx;
     93│     this._circleStyles.style.top = this._previousTop + gestureState.dy;
     94│     this._updateNativeStyles();
     95│   };
     96│



Found 24 errors
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Reply:
thus, i create CircleStyleProp type....